### PR TITLE
recursive search under quadlet admin dir and distro dir

### PR DESCRIPTION
**Pull request by harikannan512**
#### Does this PR introduce a user-facing change?

Solves #18032 

```release-note
Recursively scans directories for unit files
----------------------
debian@debian:~/Projects/podman$ bin/quadlet -dryrun
quadlet-generator[2453]: No files to parse from [/etc/containers/systemd /usr/share/containers/systemd]

----------------------
debian@debian:~/Projects/podman$ bin/quadlet -dryrun -user
quadlet-generator[2475]: Loading source unit file /home/debian/.config/containers/systemd/test_subdir/sample.container
quadlet-generator[2475]: Loading source unit file /home/debian/.config/containers/systemd/test_subdir/sub_one/sample_two.container
quadlet-generator[2475]: Loading source unit file /home/debian/.config/containers/systemd/test_subdir/sub_two/sample_one.container
---sample.service---
[Unit]
Description=The sleep container
After=local-fs.target
SourcePath=/home/debian/.config/containers/systemd/test_subdir/sample.container
RequiresMountsFor=%t/containers

[X-Container]
Image=registry.access.redhat.com/ubi9-minimal:latest
Exec=sleep 1000

[Install]
# Start by default on boot
WantedBy=multi-user.target default.target

[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
KillMode=mixed
ExecStop=/usr/local/bin/podman rm -f -i --cidfile=%t/%N.cid
ExecStopPost=-/usr/local/bin/podman rm -f -i --cidfile=%t/%N.cid
Delegate=yes
Type=notify
NotifyAccess=all
SyslogIdentifier=%N
ExecStart=/usr/local/bin/podman run --name=systemd-%N --cidfile=%t/%N.cid --replace --rm --cgroups=split --sdnotify=conmon -d registry.access.redhat.com/ubi9-minimal:latest sleep 1000

---sample_two.service---
[Unit]
Description=The sleep container
After=local-fs.target
SourcePath=/home/debian/.config/containers/systemd/test_subdir/sub_one/sample_two.container
RequiresMountsFor=%t/containers

[X-Container]
Image=registry.access.redhat.com/ubi9-minimal:latest
Exec=sleep 1000

[Install]
# Start by default on boot
WantedBy=multi-user.target default.target

[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
KillMode=mixed
ExecStop=/usr/local/bin/podman rm -f -i --cidfile=%t/%N.cid
ExecStopPost=-/usr/local/bin/podman rm -f -i --cidfile=%t/%N.cid
Delegate=yes
Type=notify
NotifyAccess=all
SyslogIdentifier=%N
ExecStart=/usr/local/bin/podman run --name=systemd-%N --cidfile=%t/%N.cid --replace --rm --cgroups=split --sdnotify=conmon -d registry.access.redhat.com/ubi9-minimal:latest sleep 1000

---sample_one.service---
[Unit]
Description=The sleep container
After=local-fs.target
SourcePath=/home/debian/.config/containers/systemd/test_subdir/sub_two/sample_one.container
RequiresMountsFor=%t/containers

[X-Container]
Image=registry.access.redhat.com/ubi9-minimal:latest
Exec=sleep 1000

[Install]
# Start by default on boot
WantedBy=multi-user.target default.target

[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
KillMode=mixed
ExecStop=/usr/local/bin/podman rm -f -i --cidfile=%t/%N.cid
ExecStopPost=-/usr/local/bin/podman rm -f -i --cidfile=%t/%N.cid
Delegate=yes
Type=notify
NotifyAccess=all
SyslogIdentifier=%N
ExecStart=/usr/local/bin/podman run --name=systemd-%N --cidfile=%t/%N.cid --replace --rm --cgroups=split --sdnotify=conmon -d registry.access.redhat.com/ubi9-minimal:latest sleep 1000
```

### CODE REVIEW

File: cmd/quadlet/main.go [hunk 1]
```diff
@@ -109,15 +109,26 @@ func getUnitDirs(user bool) []string {
 	dirs := make([]string, 0)
 	if user {
 		if configDir, err := os.UserConfigDir(); err == nil {
-			dirs = append(dirs, path.Join(configDir, "containers/systemd"))
+			dirs = returnSubPaths(dirs, path.Join(configDir, "containers/systemd"))
 		}
 	} else {
-		dirs = append(dirs, quadlet.UnitDirAdmin)
-		dirs = append(dirs, quadlet.UnitDirDistro)
+		dirs = returnSubPaths(dirs, quadlet.UnitDirAdmin)
```
Comment (rhatdan) [inline cmd/quadlet/main.go:115]: nit: Should call function appendSubPaths

**Comment by rhatdan** [comment]
I am thinking about adding a /etc/containers/systemd/user subdir which would be used by systemd to establish user quadlets across the entire system.  But I would not want these run for system quadlets.

**Comment by harikannan512** [comment]
> I am thinking about adding a /etc/containers/systemd/user subdir which would be used by systemd to establish user quadlets across the entire system. But I would not want these run for system quadlets.

Makes sense. I do have one question. Why create the user dir in `/etc/containers/systemd`. For a user quadlet run, would it not be better if the user folder was in `os.UserConfigDir()/containers/systemd`

**Comment by rhatdan** [comment]
It currently is there, but this means there is no central location for an admin to set a quadlet for all of their users.
For the use case were we want to stick login users into a container, we want to enforce that the user has no control over this and is automatically stuck inside of a container.

**Pull request closed without merging**